scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ForkedTaskVariablesManager.java

```diff
@@ -0,0 +1,157 @@
+
+        scriptHandler.addBinding(FORK_ENVIRONMENT_BINDING_NAME,
+                taskContext.getInitializer().getForkEnvironment());
+    }
+
+    public Map<String, String> extractThirdPartyCredentials(TaskContext container) throws Exception {
+        try {
+            Map<String, String> thirdPartyCredentials = new HashMap<>();
+            if (container.getDecrypter() != null) {
+                thirdPartyCredentials.putAll(container.getDecrypter().decrypt().getThirdPartyCredentials());
+            }
+            return thirdPartyCredentials;
+        } catch (Exception e) {
+            throw new Exception("Could read encrypted third party credentials", e);
+        }
+    }
+
+    public void replaceScriptParameters(Script script, Map<String, String> thirdPartyCredentials,
+            Map<String, Serializable> variables, PrintStream errorStream) {
+
+        Map<String, Serializable> variablesAndCredentials = new HashMap<>(variables);
+
+        for (Map.Entry<String, String> credentialEntry : thirdPartyCredentials.entrySet()) {
+            variablesAndCredentials.put(CREDENTIALS_KEY_PREFIX + credentialEntry.getKey(),
+                    credentialEntry.getValue());
+        }
+
+        replace(script, variablesAndCredentials, errorStream);
+    }
+
+    private void replace(Script script, Map<String, Serializable> substitutes, PrintStream errorStream) {
+        if (script != null) {
+            if ("java".equals(script.getEngineName())) {
+                try {
+                    @SuppressWarnings("unchecked")
+                    Map<String, Serializable> deserializedArgs =
+                            SerializationUtil.deserializeVariableMap(
+                                    (Map<String, byte[]>) script.getParameters()[0]);
+                    for (Map.Entry<String, Serializable> deserializedArg : deserializedArgs.entrySet()) {
+                        if (deserializedArg.getValue() instanceof String) {
+                            deserializedArg.setValue(
+                                    VariableSubstitutor.filterAndUpdate((String) deserializedArg.getValue(),
+                                            substitutes));
+                        }
+                    }
+                    script.getParameters()[0] = new HashMap<>(
+                            SerializationUtil.serializeVariableMap(deserializedArgs));
+                } catch (Exception e) {
+                    errorStream.println("Cannot read Java parameters");
+                    e.printStackTrace(errorStream);
+                }
+            } else if ("native".equals(script.getEngineName())) { // to replace script arguments
+                script.setScript(VariableSubstitutor.filterAndUpdate(script.getScript(), substitutes));
+            } else {
+                Serializable[] args = script.getParameters();
+
+                if (args != null) {
+                    for (int i = 0; i < args.length; i++) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)